torch/utils/data/dataloader.py

```diff
@@ -73,6 +77,17 @@
 
 logger = logging.getLogger(__name__)
 
+# Constants for stateful functionality
+_INDEX_SAMPLER_STATE = "_index_sampler_state"
+_SAMPLER_ITER_STATE = "_sampler_iter_state"
+_SAMPLER_ITER_YIELDED = "_sampler_iter_yielded"
+_ITERABLEDATASET_LEN_CALLED = "_IterableDataset_len_called"
+_SHARED_SEED = "_shared_seed"
+_ITERATOR_FINISHED = "_iterator_finished"
+_DATASET_STATE = "dataset_state"
+_FETCHER_STATE = "fetcher_state"
+_DATASET_ITER_STATE = "dataset_iter_state"
```

Comment:
Would maybe put these in a class, like so:
```suggestion
class _StateKeys:
    INDEX_SAMPLER_STATE = "_index_sampler_state"
    SAMPLER_ITER_STATE = "_sampler_iter_state"
    SAMPLER_ITER_YIELDED = "_sampler_iter_yielded"
    ITERABLEDATASET_LEN_CALLED = "_IterableDataset_len_called"
    SHARED_SEED = "_shared_seed"
    ITERATOR_FINISHED = "_iterator_finished"
    DATASET_STATE = "dataset_state"
    FETCHER_STATE = "fetcher_state"
    DATASET_ITER_STATE = "dataset_iter_state"
```

It's mildly annoying that some of them start with `_` and some don't.  I suppose keeping compatibility with SDL is more important than aesthetics though.